### PR TITLE
Add notice about the OTeL collector

### PR DESCRIPTION
**Pull request by tmorrisonsfx**
https://signalfuse.atlassian.net/browse/OTL-217

What changed?
- Added note that Gateway is being deprecated
- Added link to OpenTelemetry Collector Transition Guide

### CODE REVIEW

**Comment by tmorrisonsfx** [comment]
@flands @brianashby-sfx PTAL. Questions:

- Do we need more information or links?
- Is the placement of the note OK?

File: gateway/README.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ Use the SignalFx Gateway to aggregate metrics and send them to SignalFx. It is a
 **Note:** The relevant code for the project can be found in the <a target="_blank" href="https://github.com/signalfx/gateway">signalfx
 /gateway repository</a>.
 
+**Note:** The SignalFx Gateway is being deprecated. You can now deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.
```
Comment (flands) [inline gateway/README.md:61]:
```suggestion
**Note:** The SignalFx Gateway is deprecated. You should now deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.
```

File: gateway/README.md [hunk 2]
```diff
@@ -58,6 +58,8 @@ Use the SignalFx Gateway to aggregate metrics and send them to SignalFx. It is a
 **Note:** The relevant code for the project can be found in the <a target="_blank" href="https://github.com/signalfx/gateway">signalfx
 /gateway repository</a>.
 
+**Note:** The SignalFx Gateway is deprecated. You should now deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.
```
Comment (brianashby-sfx) [inline gateway/README.md:61]:
```suggestion
**Note:** The SignalFx Gateway is deprecated. Deploy the OpenTelemetry Collector and send data from the SignalFx Smart Agent to an OpenTelemetry Collector. See the [OpenTelemetry Collector Transition Guide](https://docs.signalfx.com/en/latest/otel/gateway-otel-migration.html) for more information.
```

Comment (brianashby-sfx) [reply gateway/README.md:61]: we should also link to the new deploy guide that you're working on, traci.

**Review by flands — APPROVED** [review]
LGTM

**Pull request merged**